tests/unit/s2n_tls13_handshake_state_machine_test.c

```diff
@@ -156,6 +156,42 @@ int main(int argc, char **argv)
         }
     }
 
+    /* Test: A non-FULL_HANDSHAKE form of every valid, negotiated handshake exists */
+    {
+        uint32_t handshake_type_original, handshake_type_fh;
+        message_type_t *messages_original, *messages_fh;
+
+        for (size_t i = 0; i < valid_tls13_handshakes_size; i++) {
+
+            handshake_type_original = valid_tls13_handshakes[i];
+            messages_original = tls13_handshakes[handshake_type_original];
+
+            /* Ignore INITIAL and FULL_HANDSHAKE handshakes */
+            if (!IS_NEGOTIATED(handshake_type_original) || IS_FULL_HANDSHAKE(handshake_type_original)) {
+                continue;
+            }
+
+            /* FULL_HANDSHAKE form of the handshake */
+            handshake_type_fh = handshake_type_original | FULL_HANDSHAKE;
+            messages_fh = tls13_handshakes[handshake_type_fh];
+
+            for (size_t j = 0, j_fh = 0; j < S2N_MAX_HANDSHAKE_LENGTH && j_fh < S2N_MAX_HANDSHAKE_LENGTH; j++, j_fh++) {
+                /* The original handshake cannot contain authentication messages */
+                EXPECT_NOT_EQUAL(messages_original[j], SERVER_CERT);
+                EXPECT_NOT_EQUAL(messages_original[j], SERVER_CERT_VERIFY);
+
+                 /* Skip authentication messages in the FULL_HANDSHAKE handshake */
+                 while (messages_fh[j_fh] == SERVER_CERT ||
+                         messages_fh[j_fh] == SERVER_CERT_VERIFY) {
+                     j_fh++;
```

Comment:
When `CLIENT_AUTH`, `NO_CLIENT_CERT` flags are set, the handshake is not a PSK authenticated handshake_type. We can add a unit test for this case as well.

Reply:
This test checks that every tls13 handshake that does NOT set a FULL_HANDSHAKE flag has a FULL_HANDSHAKE counterpart. We ignore the client auth handshakes in line 170 because both the CLIENT_AUTH flag and NO_CLIENT_CERT flag imply that a FULL_HANDSHAKE flag is set. So essentially I think adding extra client auth tests would redundant.

Reply:
Can we add a test to verify the assertion that `CLIENT_AUTH` flag and `NO_CLIENT_CERT` flag implies a `FULL_HANDSHAKE` flag is set?